Undo/COUndoTrackHistoryCompaction.h

```diff
@@ -0,0 +1,89 @@
+/**
+    Copyright (C) 2015 Quentin Mathe
+
+    Date:  June 2015
+    License:  MIT  (see COPYING)
+ */
+
+#import <Foundation/Foundation.h>
+#import <EtoileFoundation/EtoileFoundation.h>
+#import <CoreObject/COHistoryCompaction.h>
+
+@class COUndoTrack, COCommandGroup;
+
+/** 
+ * @group Undo
+ * @abstract A compaction strategy that targets the history located in the undo
+ * track tail or outside the undo track.
+ *
+ * @section Conceptual Model
+ *
+ * This strategy ensures that an undo track will continue to work flawlessly 
+ * for all operations (undo, redo, selective undo, set current node etc.), even 
+ * after compacting the history in the store.
+ *
+ * The undo track tail can be cut, but the remaing part up to the head will
+ * be kept intact.
+ *
+ * Revisions, persistent roots and branches referenced by other undo tracks or
+ * none are not protected by this strategy. This means you must be careful not 
+ * to break other undo tracks, when you are using multiple ones.
+ *
+ * @section Common Use Cases
+ *
+ * The most common use case is when you want to free space in a CoreObject store.
+ *
+ * Before passing the compaction strategy to -[COSQLiteStore compactHistory:], 
+ * you must always call -compute.
+ *
+ * For now, COUndoTrackHistoryCompaction is not thread-safe.
+ */
+@interface COUndoTrackHistoryCompaction : NSObject <COHistoryCompaction>
```

Comment:
So I understand this correctly, is the idea here that this object lets a COUndoTrack control the compaction of a COSQLiteStore, such that.. in GC terms, the undo track keeps alive everything that it references in the store?

I was expecting that the user would control compaction by looking at the revision graph of persistent roots, but if you aren't exposing those in a UI , and only exposing undo tracks, this makes sense I guess.